src/main/java/ch/njol/skript/patterns/TypePatternElement.java

```diff
@@ -100,19 +100,47 @@ public static TypePatternElement fromString(String s, int expressionIndex) {
 	@Nullable
 	public MatchResult match(String expr, MatchResult matchResult) {
 		int newExprOffset;
+
+		String nextLiteral = null;
+		boolean nextLiteralIsWhitespace = false;
+
 		if (next == null) {
 			newExprOffset = expr.length();
+		} else if (next instanceof LiteralPatternElement) {
+			nextLiteral = next.toString();
+
+			nextLiteralIsWhitespace = nextLiteral.trim().isEmpty();
+
+			if (!nextLiteralIsWhitespace) { // Don't do this for literal patterns that are *only* whitespace - they have their own special handling
+				// trim trailing whitespace - it can cause issues with optional patterns following the literal
+				int nextLength = nextLiteral.length();
+				for (int i = nextLength; i > 0; i--) {
+					if (nextLiteral.charAt(i - 1) != ' ') {
+						if (i != nextLength)
```

Comment:
These blocks should probably be merged.

Reply:
This is just a small check to avoid calling substring if we aren't actually getting a substring. I'm pretty sure the internal implementation checks this, but I don't think it *really* matters at this point :)